libs/perun/components/src/lib/attribute-critical-operations-toggles/attribute-critical-operations-toggles.component.ts

```diff
@@ -7,7 +7,26 @@ import { Component, EventEmitter, Input, Output } from '@angular/core';
 })
 export class AttributeCriticalOperationsTogglesComponent {
   @Input() readOperation = false;
+  @Input() readGlobal = false;
   @Input() writeOperation = true;
+  @Input() writeGlobal = false;
   @Output() readOperationChanged: EventEmitter<boolean> = new EventEmitter<boolean>();
+  @Output() readGlobalChanged: EventEmitter<boolean> = new EventEmitter<boolean>();
   @Output() writeOperationChanged: EventEmitter<boolean> = new EventEmitter<boolean>();
+  @Output() writeGlobalChanged: EventEmitter<boolean> = new EventEmitter<boolean>();
+
+  changeOperation(
+    operationChanged: EventEmitter<boolean>,
+    operation: boolean,
+    globalChanged: EventEmitter<boolean>,
+    global: string
+  ): void {
+    operationChanged.emit(!operation);
+
+    // if the operation is changing from true to false, then we want to change also global criticality to false
+    if (operation) {
+      this[global] = false;
+      globalChanged.emit(false);
+    }
```

Comment:
I understand you wanted to keeps things DRY but I am quite against `this[global]` for multiple reasons. Firstly as a caller of this method you don't get any help from Typescript about what values of `global` are valid and need to check manually. This also means there is no warning that you are assigning to non existent property or property of different type. Lastly any change to variable names requires change to all the method calls again with no warning.
Right now I see two approaches:
- Better to repeat yourself than shoot yourself in the foot. Just create two separate methods for write and read. Straight forward option.
- Instead of having output (maybe even input) for each property have one output with all values wrapped in an object. May be a little bit less cluttered. This would however, mean that every time all values need to be reassigned (object created, ...) which is wasteful. Although here it should not be a problem.